File: src/Dto/SseSnapshot/Infer-task-snapshot.ts
/**
 * SSE 推理任务快照（event: model:inference_event:first / model:inference_event）
 */
export interface InferTaskSnapshot {
    //  任务id
    taskId: number
    // 当前已经处理评论数 卡片上展示
    processedCount: number
    // 推理持续时间 单位秒
    duration: number
    // 当前时间 eg"2026-02-14 20:20:47", 卡片上展示
    currentTime: string
    // 状态代码 0:处理中 1:成功完成 2:异常完成
    status: number
    // 状态描述，用于在卡片上展示 处理中、成功完成、异常完成
    statusMsg: string
    //   平均处理速度
    processSpeed: number | null
}

File: Dto/Redis/InferTaskSnapshot.py
import json
from dataclasses import dataclass, asdict
from datetime import datetime

#todo  统一status的含义与springboot后台的含义
@dataclass
class InferTaskSnapshot:
    taskId: int
    processedCount: int
    duration: int #单位为秒
    status: int
    statusMsg: str
    currentTime: str = None
    processSpeed: float = 0.0

    def __post_init__(self):
        # 初始化时间并处理浮点数精度
        if not self.currentTime:
            self.currentTime = datetime.now().strftime("%Y-%m-%d %H:%M:%S")

    def to_json(self) -> str:
        return json.dumps(asdict(self), ensure_ascii=False)

# 使用方法：
# snapshot = TaskSnapshot(task_id, count, duration, status, msg)
# payload_json = snapshot.to_json()
# snapshot = replace(old_snapshot, currentTime=None)

